common/templates/context_funcs.go

```diff
@@ -518,6 +518,43 @@ func (c *Context) tmplPinMessage(unpin bool) func(channel, msgID interface{}) (s
 	}
 }
 
+func (c *Context) tmplSuppressEmbeds(suppress bool) func(channel, msgID interface{}) (string, error) {
+	return func(channel, msgID interface{}) (string, error) {
+		if c.IncreaseCheckGenericAPICall() {
+			return "", ErrTooManyAPICalls
+		}
+
+		cID := c.ChannelArgNoDM(channel)
+		if cID == 0 {
+			return "", errors.New("unknown channel")
+		}
+
+		mID := ToInt64(msgID)
+
+		message, _ := common.BotSession.ChannelMessage(cID, mID)
+		if message == nil {
+			return "", errors.New("unknown message")
+		}
+
+		newFlags := message.Flags | discordgo.MessageFlagsSuppressEmbeds
+		if !suppress {
+			newFlags = newFlags ^ discordgo.MessageFlagsSuppressEmbeds
```

Comment:
Why XOR here instead of OR?

Reply:
The intent is for L539 to ensure the presence of the flag. While L541 executes if the function is set to unsuppress to then remove the flag.